lib/Test/Nginx/Socket.pm

```diff
@@ -2254,24 +2280,23 @@ sub gen_curl_cmd_from_req ($$) {
 
     for my $h (@headers) {
         #warn "h: $h\n";
-        if ($h =~ /^\s*User-Agent\s*:\s*(.*\S)/i) {
-            push @args, '-A', $1;
-
-        } else {
-            if ($h =~ /^\s*Content-Type\s*:/i) {
-                $found_content_type = 1;
-            }
-
-            push @args, '-H', $h;
+        if ($h =~ /^\s*Content-Type\s*:/i) {
+            $found_content_type = 1;
         }
+
+        push @args, '-H', $h;
     }
 
     if ($req =~ m{\G(.+)}gcsm) {
         #warn "!! POST body data len: ", length($1);
         if (!$found_content_type) {
             push @args, "-H", 'Content-Type: ';
         }
-        push @args, '--data-binary', $1;
```

Comment:
We should assign the match result to a local variable, if it is not used immediately.

```
my $body = $1;
```